### PR TITLE
[ISSUE-766] Remove replace directives

### LINKED ISSUE #766: Make csi-driver go.mod importable and remove replace directive as much as possible
**Is your feature request related to a problem? Please describe.**

Remove replace directive from sources

### PR DESCRIPTION
**Pull request by rkiselenko**
### Problem

Make csi-driver go.mod importable and remove replace directive as much as possible

In order to remove all replace we should remove deps which not related to actual driver. 

k8s e2e modules isn't importable and imported with replace directive.

### Resolution

Move all e2e tests from folder `test` to its own module `csi-baremetal-e2e-tests` and require `csi-baremetal` module as deps for it.

Cleanup deps and run `go mod tidy` remove all indirect deps.

## PR checklist
- [x] Add link to the issue
- [x] Choose Project
- [x] Choose PR label
- [ ] New unit tests added
- [ ] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

## Testing
_Provide test details_

### CODE REVIEW

File: tests/go.mod [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+	github.com/onsi/gomega v1.14.0
+	github.com/sirupsen/logrus v1.8.1
+	gopkg.in/yaml.v2 v2.4.0
+	k8s.io/api v1.22.5
+	k8s.io/apimachinery v0.23.4
+	k8s.io/client-go v0.23.4
+	k8s.io/kubernetes v1.22.5
+)
+
+require (
```
Comment (safronovD) [inline tests/go.mod:21]: Why we have 2 `require` sections here?

File: go.mod [hunk 2]
```diff
@@ -5,63 +5,25 @@ go 1.16
 require (
```
Comment (safronovD) [inline go.mod:5]: Should we change go version here too?

File: tests/e2e/reports/report.xml [hunk 3]
```diff
@@ -0,0 +1,543 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (yurkov-anton) [inline tests/e2e/reports/report.xml:1]: Is this file need to merge PR?

**Review by yurkov-anton — APPROVED** [review]

File: tests/go.mod (hunk 1 above)
Comment (rkiselenko) [reply tests/go.mod:21]: idk go mod works in that way

File: go.mod (hunk 2 above)
Comment (rkiselenko) [reply go.mod:5]: No

File: tests/go.mod (hunk 1 above)
Comment (safronovD) [reply tests/go.mod:21]: thank you for the reference

File: go.mod (hunk 2 above)
Comment (safronovD) [reply go.mod:5]: So we have go 1.16 in devkit and in csi-beremetal go mod, go 1.17 in PR validation and in csi-baremetal-tests go.mod. Why do we need to upgrade go version for tests and why can't update it everywhere?

Comment (rkiselenko) [reply go.mod:5]: `go help go.mod` 

> The expected language version, set by the go directive, determines
> which language features are available when compiling the module.
> Language features available in that version will be available for use.
> Language features removed in earlier versions, or added in later versions,
> will not be available. Note that the language version does not affect
> build tags, which are determined by the Go release being used.

Comment (rkiselenko) [reply go.mod:5]: > The basic guideline is fairly simple: a "go" directive 1.N means that the code in that module is permitted to use language features that existed in 1.N even if those features were removed in later releases of the language.

Comment (safronovD) [reply go.mod:5]: Ok, could you please explain why do we need to upgrade go version for csi-baremetal-tests only? Is it works in devkit with go 1.16?

Comment (rkiselenko) [reply go.mod:5]: There is no "version upgrade", `go 1.N` declare the version as described above, you free to use any version.

Comment (safronovD) [reply go.mod:5]: But we use go compailer 1.16 installed in devkit and declare go 1.17 in go.mod. May we have conflicts?

Comment (rkiselenko) [reply go.mod:5]: no

**Review by safronovD — APPROVED** [review]

File: Makefile [hunk 4]
```diff
@@ -92,33 +92,36 @@ install-protoc:
 install-compile-proto: install-protoc compile-proto
 
 install-controller-gen:
-	# Build controller-gen from go.mod
```
Comment (safronovD) [inline Makefile:95]: Could you please remove `tidy` target too?

Comment (rkiselenko) [reply Makefile:95]: I dont get.

**Pull request merged**